### DISCUSSION TITLE
Creating Delta<T> from existing object for unit testing purposes

### DISCUSSION THREAD

**Discussion by groogiam** [OData/AspNetCoreOData#486] [Q&A]
Is it possible to create an instance of Delta from an existing object for unit testing purposes? I have some business rule logic classes called from my controller that take the Delta coming from the patch operations as an input and I'm having trouble figure out how to create a mock instance of Delta to properly unit test them. Thanks for you help.

**Comment by julealgon** [comment]
Last time I had to deal with unit testing code that relied on `Delta<T>` I created my own `IDelta<T>` abstraction and an adapter for the real delta so I could test everything.

I know this doesn't necessarily answer your question but it might be of use to you. In most cases, you are only interested in testing the interaction with the `Patch/Put` methods in the delta object and those can be mocked easily with an adapter+interface approach.
